docs/user/kafka/confluent.rst

```diff
@@ -7,11 +7,189 @@ Confluent Integration
+``iso-date``                ``STRING``            Date format ``yyyy-MM-dd``
+``iso-instant``             ``STRING``            Date format ``yyyy-MM-dd'T'HH:mm:ss.SSSZZ``
+``iso-instant-no-millis``   ``STRING``            Date format ``yyyy-MM-dd'T'HH:mm:ssZZ``
+=========================== ===================== ====================================================
+
+``geomesa.date.default``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Indicates that the field represents the default ``Date`` for this ``SimpleFeatureType``. If the key
+``geomesa.date.format`` is not present on the same schema field, this attribute will be ignored.
+
+=========== =============================
+Value       Description
+=========== =============================
+``true``    The default ``Date``
+``false``   Not the default ``Date``
+=========== =============================
+
+``geomesa.index``
+^^^^^^^^^^^^^^^^^
+
+Indicates that the field should be indexed by GeoMesa.
+
+=========== ==================================
+Value       Description
+=========== ==================================
+``true``    Index this attribute
+``false``   Do not index this attribute
+``full``    Index the full ``SimpleFeature``
+=========== ==================================
+
+``geomesa.cardinality``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Specifies a cardinality hint for the attribute, for use by GeoMesa in query planning. This property should be used
+in conjunction with ``geomesa.index``.
+
+=========== ====================================
+Value       Description
+=========== ====================================
+``high``    Prioritize this attribute index
+``low``     De-prioritize this attribute index
+=========== ====================================
+
+``geomesa.avro.visibility.field``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Specifies a field in the Avro schema which should be used as the visibility for features of the ``SimpleFeatureType``.
+
+============= ===================== ========================================================
+Value         Schema Field Type     Description
+============= ===================== ========================================================
+``true``      ``STRING``            Use the value of this field as the feature visibility
+``false``     ``STRING``            Do not use this field as the feature visibility
+============= ===================== ========================================================
+
+Example Schema
+--------------
+
+Any additional top-level properties on the schema will be included in the SFT as user data.
+
+::
+
+    {
+      "namespace": "org.locationtech",
+      "type": "record",
+      "name": "SchemaRegistryMessage",
+      "geomesa.table.sharing" = "false",
+      "geomesa.table.compression.enabled" = "true",
+      "fields": [
+        {
+          "name": "id",
+          "type": "string",
+          "geomesa.index": "full",
+          "geomesa.cardinality": "high"
```

Comment:
I do like the indexing hints here.  This opens up some really good ideas.  It also could get us into interesting situations going forward with compatibility.  

That is, how are we going to use this downstream?

Also, if we are hinting this much in the Avro schema, it may make sense to have a way to specify "live"/streaming FIDs and data-dependent FIDs for Accumulo/HBase.

Reply:
I wasn't originally going to include any of the ``geomesa.index`` or ``geomesa.cardinality`` stuff, but I saw that some of our SFTs specified them so I wanted to make sure it was compatible.

Reply:
Removed